server/routerlicious/packages/lambdas/src/deli/lambdaFactory.ts

```diff
@@ -164,14 +164,42 @@ export class DeliLambdaFactory extends EventEmitter implements IPartitionLambdaF
             this.serviceConfiguration,
             sessionMetric,
             sessionStartMetric);
+
+        deliLambda.on("close", (closeType) => {
+            const handler = async () => {
+                if ((closeType === LambdaCloseType.ActivityTimeout || closeType === LambdaCloseType.Error)) {
+                    const result = await this.collection.findOne({ documentId });
```

Comment:
Instead of querying and then updating, we should be able to update the isSessionAlive for the docId in one call itself: https://stackoverflow.com/questions/10290621/how-do-i-partially-update-an-object-in-mongodb-so-the-new-object-will-overlay

Reply:
I try some times, and it failed
![Screen Shot 2022-04-04 at 5 31 10 PM](https://user-images.githubusercontent.com/66701969/161655246-909b5be4-eb9a-438e-a3ce-86a0aaa59370.png)

Maybe let's keep it as it is now.

Reply:
Let's try not to postpone things are in the critical path that could affect latency or reliability. Since, we are using the update() in our services-core database.ts and not the mongo update(), we might have to update without using the $set. Can you please try replacing line 72 with,

`const data = { "session.isSessionAlive": "false" }`

Example: https://github.com/microsoft/FluidFramework/blob/fefc3ae5831d7f9cf9a9cee09a8e02b4101f801b/server/routerlicious/packages/test-utils/src/testHistorian.ts#L194

Reply:
This works now. Thanks for sharing!